File: EXPERIENCIA-DE-PRÁCTICA-1-ABSTRACCIÓN/EJERCICIO-1.2/clase_coche.java
// Clase Coche: Representa un coche con atributos como modelo, año y velocidad, y métodos para interactuar con el coche. 

public class Coche { 

    // Atributos 

    private String modelo; 

    private int año; 

    private int velocidad; 

 

    // Constructor 

    public Coche(String modelo, int año, int velocidad) { 

        this.modelo = modelo; 

        this.año = año; 

        this.velocidad = velocidad; 

    } 

 

    // Métodos para acceder a los atributos 

    public String getModelo() { // Modelo del coche 

        return modelo; 

    } 

 

    public int getAño() { // Año del coche 

        return año; 

    } 

 

    public int getVelocidad() { // Velocidad actual del coche 

        return velocidad; 

    } 

 

    // Método para acelerar el coche 

    public void acelerar() { 

        velocidad += 1; // Aumenta la velocidad en 1 km/h 

        System.out.println(modelo + " ha acelerado. Velocidad actual: " + velocidad + " km/h"); 

    } 

 

    // Método para frenar el coche 

    public void frenar() { 

        if (velocidad >= 80) { 

            velocidad -= 10; // Disminuye la velocidad en 10 km/h 

            System.out.println(modelo + " ha frenado. Nueva velocidad: " + velocidad + " km/h"); 

        } else { 

            System.out.println(modelo + " siga, tiene una velocidad adecuada. Velocidad actual: " + velocidad + " km/h"); 

        } 

    } 

 

    // Método main para probar la clase 

    public static void main(String[] args) { 

        // Creación de diferentes objetos para representar diferentes coches 

        Coche coche1 = new Coche("Toyota Corolla", 2020, 60); 

        Coche coche2 = new Coche("Ford Mustang", 2018, 120); 

        Coche coche3 = new Coche("Tesla Model S", 2022, 100); 

 

        // Interacción 

        coche1.acelerar(); 

        coche1.frenar(); 

 

        coche2.acelerar(); 

        coche2.frenar(); 

 

        coche3.acelerar(); 

        coche3.frenar(); 

    } 

} 

File: EXPERIENCIA-DE-PRÁCTICA-1-ABSTRACCIÓN/EJERCICIO-1.2/ClaseCoche.cpp
#include <iostream>
#include <string>

using namespace std;

// clase Coche
class Coche {
private:
    string modelo;
    int año;
    double velocidad;

public:
    // valores de los atributos
    void setModelo(string _modelo) // modelo del coche
    {
        modelo = _modelo;
    }

    void setAño(int _año) // año del coche
    {
        año = _año;
    }

    void setVelocidad(double _velocidad) // la velocidad del coche
    {
        velocidad = _velocidad;
    }

    // obtener los valores de los atributos
    string getModelo() const // modelo del coche
    {
        return modelo;
    }

    int getAño() const { // año del coche
        return año;
    }

    double getVelocidad() const { // velocidad del coche
        return velocidad;
    }

    // acciones con el coche
    void acelerar(double aumento) { // la velocidad del coche
        velocidad += aumento;
        cout << "El coche esta acelerando---> Velocidad actual---> " << velocidad << " km/h" << endl;
    }

    void frenar(double reduccion) { // reducir la velocidad del coche
        if (velocidad >= reduccion) {
            velocidad -= reduccion;
            cout << "El coche esta frenando---> Velocidad actual---> " << velocidad << " km/h" << endl;
        } else {
            cout << "El coche ya está detenido." << endl;
        }
    }
};

int main() {

    Coche coche1; // Instancia de un coche
    coche1.setModelo("Toyota Corolla"); //  modelo del coche
    coche1.setAño(2022); // año del coche
    coche1.setVelocidad(60); // velocidad inicial del coche

    Coche coche2; // Instancia de otro coche
    coche2.setModelo("Honda Civic"); // modelo del otro coche
    coche2.setAño(2020); // año del otro coche
    coche2.setVelocidad(52); // velocidad inicial del otro coche

    // Mostrar información y realizar acciones sobre los coches
    cout <<"\nCoche 1:" <<endl; // información sobre el coche 1
    cout << "Modelo ---> " << coche1.getModelo() << "\nAnio ---> " << coche1.getAño() << "\nVelocidad ---> " << coche1.getVelocidad() << " km/h" << endl; // Imprimir modelo, año y velocidad del coche 1
    coche1.acelerar(50); // Acelerar el coche 1
    coche1.frenar(20); // Frenar el coche 1

    cout <<"\nCoche 2:" <<endl; // información sobre el coche 2
    cout << "Modelo ---> " << coche2.getModelo() << "\nAnio ---> " << coche2.getAño() << "\nVelocidad ---> " << coche2.getVelocidad() << " km/h" << endl; // Imprimir modelo, año y velocidad del coche 2
    coche2.acelerar(70); // Acelerar el coche 2
    coche2.frenar(80); // Frenar el coche 2

    return 0;
}
